code-challenges.test.js

```diff
@@ -40,11 +61,27 @@ var hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
 var hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
 // Expected output: [ 2, 1, -1 ]
 
-
+describe("remainder", () => {
+  test("takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3", () => {
+    expect(remainder(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+    expect(remainder(hodgepodge2)).toEqual([ 2, 1, -1 ])
+  })
+})
 
 // b) Create the function that makes the test pass.
 
+//declare a function called remainder which takes an array argument
+//use filter method to return an array with only numbers
+//
 
+const remainder = (array) => {
+  let newArray = array.filter(value => {
+    return typeof value === 'number'
+  })
+  return newArray.map(value => {
+    return value % 3
+  })
+}
```

Comment:
Good, but there is a way to reduce the processing power from n^4 to n^2. By using "&&" in the initial return.